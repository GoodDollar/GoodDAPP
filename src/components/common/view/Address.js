// @flow
import React from 'react'
import Text from './Text'

type AddressProps = {
  value: string
}

const Address = React.memo(({ value }: AddressProps) => (
  <Text>{value && `${value.slice(0, 6)}...${value.slice(value.length - 4)}`}</Text>
))

export default Address
