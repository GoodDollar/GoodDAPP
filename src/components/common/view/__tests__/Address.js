import React from 'react'
import renderer from 'react-test-renderer'
import Address from '../Address'

// Note: test renderer must be required after react-native.

describe('Address', () => {
  it('matches snapshot', async () => {
    let component

    // eslint-disable-next-line require-await
    await renderer.act(async () => (component = renderer.create(<Address />)))
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
