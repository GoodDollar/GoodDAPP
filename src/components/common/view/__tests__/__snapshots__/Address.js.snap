// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address matches snapshot 1`] = `
<div
  className="css-text-901oao"
  dir="auto"
/>
`;
